pages/[username]/[collectionId]/[tokenId]/index.tsx

```diff
@@ -19,23 +20,27 @@ type NftDetailPageProps = MetaTagProps & {
 };
 
 export default function NftDetailPage({ username, collectionId, tokenId }: NftDetailPageProps) {
-  const { push } = useRouter();
+  const { push, query } = useRouter();
 
   // the default "back" behavior from the NFT Detail Page
-  // is a redirect to the Collection Page
-  const collectionRoute = useMemo(
-    () => ({
+  // is a redirect to the Collection Page or a valid returnTo route
+  const returnToRoute = useMemo<Route>(() => {
+    if (query?.returnTo === 'home') {
+      return {
+        pathname: '/home',
+      };
+    }
+    return {
       pathname: '/[username]/[collectionId]',
       query: { username, collectionId },
-    }),
-    [username, collectionId]
-  );
+    };
+  }, [username, collectionId, query?.returnTo]);
```

Comment:
Q: should the previous route just specify the entire `returnTo`?

e.g. set `returnTo` literally as `/home` or `/evgency/12345-6789`?

typescript may not be happy but could help with co-location. then we don't even need this `returnToRoute` logic

Reply:
Yeah I kinda did it to make sure you only pass valid paths. But yeah, I could make sure the previous page passes a query, but also - we prob need a fallback? If so, it prob won't change the logic that much.

Essentially we will have something like

```
  const returnToRoute = useMemo<Route>(() => {
    if (query?.returnTo != null) {
      return {
        pathname: query.returnTo,
      };
    }
    return {
      pathname: '/[username]/[collectionId]',
      query: { username, collectionId },
    };
  }, [username, collectionId, query?.returnTo]);
  ```

Reply:
ahh i gotchu, makes sense. okay i think this all relates to [this comment](https://github.com/gallery-so/gallery/pull/1179/files#r1041157039) further below (read it first!)

we have a [hook](https://github.com/gallery-so/gallery/blob/3dc745d80b407579db61e8f80ea6f68ed351c8ca/src/scenes/NftDetailPage/useDisplayFullPageNftDetailModal.tsx#L12) that we need to use to make the modal work, and it has a similar `returnTo` logic. i think we could refactor all that and have the user always pass in a `returnTo` prop into the hook!

in fact maybe we could co-locate / encapsulate everything within the `LinkToNftDetailView` component (described in the comment). so basically the component itself could take a `returnTo` prop, use the hook, etc.